src/ObservationDialog/Select.js

```diff
@@ -101,13 +116,14 @@ export const SelectMultiple = ({
   ...props
 }: Props) => {
   const classes = useStyles()
+  const encoder = Encoder(options)
   return (
     <Autocomplete
       id={id}
       multiple
       freeSolo
-      value={value || []}
-      onChange={(e, v) => onChange(v)}
+      value={(value || []).map(encoder.toLabel)}
+      onChange={(e, v) => onChange(v.map(encoder.toValue))}
       options={options.map(op => (typeof op === 'object' ? op.label : op))}
```

Comment:
I missed this before but flow typecheck caught it. `typeof null === 'object'` because JavaScript... hence
 
```suggestion
      options={options.map(op => (op && typeof op.label === 'string' ? op.label : op))}
```